packages/website/src/components/Playground.tsx

```diff
@@ -70,6 +70,34 @@ function Playground(): JSX.Element {
   const [position, setPosition] = useState<Monaco.Position | null>(null);
   const [activeTab, setTab] = useState<TabType>('code');
   const [showModal, setShowModal] = useState<TabType | false>(false);
+  const [editorWidth, setEditorWidth] = useState<number | undefined>();
+  const [dragging, setDragging] = useState<boolean>(false);
+  const [editorLeftEdge, setEditorLeftEdge] = useState<number | undefined>();
+  const sourceCodeContainerRef = React.createRef<HTMLDivElement>();
+
+  useEffect(() => {
+    const handleMouseMove = (e: MouseEvent): void => {
+      if (dragging && editorLeftEdge) {
+        setEditorWidth(e.clientX - editorLeftEdge);
+      }
+    };
+    const handleMouseUp = (): void => {
+      setDragging(false);
+    };
+
+    const boundingRect =
+      sourceCodeContainerRef.current?.getBoundingClientRect();
+    setEditorLeftEdge(boundingRect?.x);
+    setEditorWidth(boundingRect?.width);
+
+    window.addEventListener('mousemove', handleMouseMove);
+    window.addEventListener('mouseup', handleMouseUp);
+
+    return (): void => {
+      window.removeEventListener('mousemove', handleMouseMove);
+      window.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [sourceCodeContainerRef]);
```

Comment:
This works pretty well for mouse movements, nicely done!

But, there are a bunch of libraries out there that do this work for us. I've personally used [`react-split-pane`](https://github.com/tomkp/react-split-pane) and liked it. Have you considered using one of them?

It'd be nice to take in a well-tested library instead of writing the code ourselves. There are almost certainly edge cases in this UI behavior that'll pop up as bugs in the future. And we'll probably want to make the left sidebar resizeable too.

Reply:
Thanks! The reason I didn't use a library was to avoid the overhead of another dependency for this one use case. But I agree it would be more robust.

Are there any criteria for adding dependencies, such as package size or activeness? `react-split-pane` looks very usable but I wanted to confirm before installing (it's version 0.1.92 and last published 2 years ago. Whether this is matters can vary from project to project).

Reply:
Funny, react-split-pane was one of the more recently active ones the last time I investigated it... which was about 2 years ago 😄 

As long as the package works and isn't ridiculously larger than others, I don't think we have any hard dependencies. If you want to do due diligence and post a comparison of size & maintenance level across packages that could be helpful?

Reply:
I did some searching and it looks like the two best options are react-split-pane and [Split.js](https://split.js.org/#/split-grid). For Split.js, we would want to use the [React Split Grid](https://www.npmjs.com/package/react-split-grid) wrapper since Split.js is by default a vanilla JavaScript library.
||[react-split-pane](https://tomkp.github.io/react-split-pane/?path=/story/basic--basic-horizontal)|[Split.js](https://split.js.org/#/split-grid)|
|---|---|---|
|Pros|<ul><li>Simpler API/better abstraction: wrap panels in `<SplitPane></SplitPane>` and it's good to go</li><li>Many downloads on npm: 243,817 / week</li></ul>|<ul><li>Long history: maintained since 2014</li><li>Used in production by sites like [JSFiddle](https://jsfiddle.net/)</li><li>Smaller bundle size: 4.1kB minified + gzipped</li></ul>|
|Cons|<ul><li>Possibly less robust: latest version is 0.1.92</li><li>Larger bundle size: 9.3kB minified + gzipped</li></ul>|<ul><li>Harder to use: have to call implementation details such as `getGridProps` and `getGutterProps`</li></ul>|

Bundle sizes are from [https://bundlephobia.com](https://bundlephobia.com).

It seems `react-split-pane` is easier to use but comes with a larger package size (it's also less maintained, though this might not be an issue since the UI of dragging/resizing doesn't change over time). What are your thoughts?

Reply:
> easier to use

That's probably my vote then 😄. This is the first version of the playground; it's only a matter of time before we rewrite the whole thing (as with any v1).

But if you have a strong preference, go for it!

Reply:
Sounds good.

I tried to install `react-split-pane` but its React peer dependency is major version 16 while the typescript-eslint's is version 18:
```sh
npm ERR! Found: react@18.2.0
...
npm ERR! peer react@"^16.0.0-0" from react-split-pane@0.1.92
```
Should we pass `--legacy-peer-deps` to solve? Or avoid using this package.

Reply:
Can you use `resolutions` in the `package.json` to ignore this? IIRC I've had to do that before.

Reply:
Yes, this works. Thanks!